src/lightning_app/cli/lightning_cli.py

```diff
@@ -337,6 +339,75 @@ def stop() -> None:
 _main.add_command(create)
 
 
+@_main.command("ssh")
+@click.option(
+    "--app-id",
+    "app_id",
+    type=str,
+    default=None,
+    required=False,
+)
+@click.option(
+    "--component-name",
+    "component_name",
+    type=str,
+    default=None,
+    help="Specify which component to SSH into",
+)
+def ssh(app_id: str = None, component_name: str = None) -> None:
+    """SSH into a Lighting App."""
+    import inquirer
+
+    app_manager = _AppManager()
+    if app_id is None:
+        apps = app_manager.list_apps(phase_in=[V1LightningappInstanceState.RUNNING])
+        if len(apps) == 0:
+            raise click.ClickException("no running apps available.")
+
+        available_apps = [
+            inquirer.List(
+                "app_name",
+                message="What app to SSH into?",
+                choices=[app.name for app in apps],
+            ),
+        ]
+        app_name = inquirer.prompt(available_apps)["app_name"]
+        app_id = [app.id for app in apps if app.name == app_name][0]
+
+    try:
+        instance = app_manager.get_app(app_id=app_id)
+    except ApiException:
+        raise click.ClickException("failed fetching app instance")
+
+    components = app_manager.list_components(app_id=app_id)
+    if component_name is None:
+        available_components = [
+            inquirer.List(
+                "component_name",
+                message="Which component to SSH into?",
+                choices=[work.name for work in components] + ["flow"],
+            )
+        ]
+        component_name = inquirer.prompt(available_components)["component_name"]
+
+    component_id = None
+    if component_name == "flow":
+        component_id = f"lightningapp-{app_id}"
+    elif component_name is not None:
+        work_id = next(iter([work.id for work in components if work.name == component_name]), None)
+        if work_id is not None:
+            component_id = f"lightningwork-{work_id}"
+
+    if component_id is None:
+        raise click.ClickException(f"unable to find app component with name {component_name}")
+
+    app_cluster = app_manager.get_cluster(cluster_id=instance.spec.cluster_id)
+    ssh_endpoint = app_cluster.status.ssh_gateway_endpoint
+
+    os.execv("/usr/bin/ssh", ["-tt", f"{component_id}@{ssh_endpoint}"])
```

Comment:
Could we do some research how to find the `ssh` command? IMO, we should at least check if the ssh command is available. For example with: `ssh_available = subprocess.call(["command", "-v", "ssh"]) == 0`.

Reply:
good point, I meant to switch this over to [shutil](https://docs.python.org/3/library/shutil.html)